devolutions-crypto-wayk/src/bastion/mod.rs

```diff
@@ -0,0 +1,56 @@
+/// Primitives for Wayk Bastion Key derivation
+pub mod key_derive;
+
+/// Primitives for Wayk Bastion Key Exchange protocol
+pub mod key_exchange;
+
+/// Primitives for Field-Level Encryption (FLE)
+pub mod fle;
+
+use std::error::Error as StdError;
+use std::fmt;
+use uuid::Uuid;
+
+/// Bastion crypto error
+#[derive(Debug)]
+pub enum Error {
+    XChaCha20,
+    InvalidSize { got: usize },
+    Base64(base64::DecodeError),
+    Rand(rand::Error),
+    MasterKeyIdMismatch { expected: Uuid, found: Uuid },
+    Padding,
```

Comment:
The padding error should never happen technically and MUST be hidden from the other party(AKA return the same error to the client if the tag validation failed or if the padding is invalid. This is the basis of padding oracles, even though AFAIK this error should never happen because of the tag validation

Reply:
Okay, I'll remove this variant and just `.expect(…)` instead because as you said it should not happen.